learning_resources/migrations/0043_add_nestable_topics.py

```diff
@@ -0,0 +1,37 @@
+# Generated by Django 4.2.11 on 2024-04-29 19:31
+
+import django.db.models.deletion
+import django.db.models.functions.text
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("learning_resources", "0042_update_ondelete_for_events"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="learningresourcetopic",
+            name="parent",
+            field=models.ForeignKey(
+                blank=True,
+                null=True,
+                on_delete=django.db.models.deletion.CASCADE,
+                to="learning_resources.learningresourcetopic",
+            ),
+        ),
+        migrations.AlterField(
+            model_name="learningresourcetopic",
+            name="name",
+            field=models.CharField(max_length=128),
+        ),
+        migrations.AddConstraint(
```

Comment:
This migration is not reversable 
abeglova@Anastasias-MacBook-Pro-2 mit-open % docker-compose run web ./manage.py migrate learning_resources 0042
[+] Creating 3/0
 ✔ Container mit-open-db-1            Running                                                                                        0.0s
 ✔ Container mit-open-redis-1         Running                                                                                        0.0s
 ✔ Container opensearch-node-mitopen  Running                                                                                        0.0s
Operations to perform:
  Target specific migration: 0042_update_ondelete_for_events, from learning_resources
Running migrations:
  Rendering model states... DONE
  Unapplying learning_resources.0043_add_nestable_topics...Traceback (most recent call last):
  File "/opt/venv/lib/python3.11/site-packages/django/db/backends/utils.py", line 87, in _execute
    return self.cursor.execute(sql)
           ^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.errors.UniqueViolation: could not create unique index "learning_resources_learningresourcetopic_name_2a9dc4dd_uniq"
DETAIL:  Key (name)=(Innovation) is duplicated.


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/src/./manage.py", line 14, in <module>
    execute_from_command_line(sys.argv)
  File "/opt/venv/lib/python3.11/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/opt/venv/lib/python3.11/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/opt/venv/lib/python3.11/site-packages/django/core/management/base.py", line 412, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/opt/venv/lib/python3.11/site-packages/django/core/management/base.py", line 458, in execute
    output = self.handle(*args, **options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/django/core/management/base.py", line 106, in wrapper
    res = handle_func(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/django/core/management/commands/migrate.py", line 356, in handle
    post_migrate_state = executor.migrate(
                         ^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/django/db/migrations/executor.py", line 141, in migrate
    state = self._migrate_all_backwards(plan, full_plan, fake=fake)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/django/db/migrations/executor.py", line 219, in _migrate_all_backwards
    self.unapply_migration(states[migration], migration, fake=fake)
  File "/opt/venv/lib/python3.11/site-packages/django/db/migrations/executor.py", line 279, in unapply_migration
    state = migration.unapply(state, schema_editor)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/django/db/migrations/migration.py", line 193, in unapply
    operation.database_backwards(
  File "/opt/venv/lib/python3.11/site-packages/django/db/migrations/operations/fields.py", line 240, in database_backwards
    self.database_forwards(app_label, schema_editor, from_state, to_state)
  File "/opt/venv/lib/python3.11/site-packages/django/db/migrations/operations/fields.py", line 235, in database_forwards
    schema_editor.alter_field(from_model, from_field, to_field)
  File "/opt/venv/lib/python3.11/site-packages/django/db/backends/base/schema.py", line 831, in alter_field
    self._alter_field(
  File "/opt/venv/lib/python3.11/site-packages/django/db/backends/postgresql/schema.py", line 288, in _alter_field
    super()._alter_field(
  File "/opt/venv/lib/python3.11/site-packages/django/db/backends/base/schema.py", line 1094, in _alter_field
    self.execute(self._create_unique_sql(model, [new_field]))
  File "/opt/venv/lib/python3.11/site-packages/django/db/backends/postgresql/schema.py", line 48, in execute
    return super().execute(sql, None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/django/db/backends/base/schema.py", line 201, in execute
    cursor.execute(sql, params)
  File "/opt/venv/lib/python3.11/site-packages/django/db/backends/utils.py", line 102, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/sentry_sdk/integrations/django/__init__.py", line 644, in execute
    result = real_execute(self, sql, params)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/django/db/backends/utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "/opt/venv/lib/python3.11/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/opt/venv/lib/python3.11/site-packages/django/db/backends/utils.py", line 87, in _execute
    return self.cursor.execute(sql)
           ^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.IntegrityError: could not create unique index "learning_resources_learningresourcetopic_name_2a9dc4dd_uniq"
DETAIL:  Key (name)=(Innovation) is duplicated.